src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -247,4 +238,100 @@ public boolean verify(String hostname, SSLSession session) {
     };
     HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
   }
+
+  private boolean executeHTTPServiceAndCheckStatusCode() throws IOException {
+    HttpURLConnection conn = null;
+
+    Map<String, String> headers = config.getRequestHeadersMap();
+
+    if (accessToken == null || OAuthUtil.tokenExpired(accessToken)) {
+      accessToken = OAuthUtil.getAccessToken(config);
+    }
+
+    if (accessToken != null) {
+      Header authorizationHeader = new BasicHeader("Authorization",
+                                                   String.format("Bearer %s", accessToken.getTokenValue()));
+      headers.putAll(config.getHeadersMap(String.valueOf(authorizationHeader)));
+    }
+
+    try {
+      URL url = new URL(configURL);
+      conn = (HttpURLConnection) url.openConnection();
+      if (conn instanceof HttpsURLConnection) {
+        //Disable SSLv3
+        System.setProperty("https.protocols", "TLSv1,TLSv1.1,TLSv1.2");
+        if (config.getDisableSSLValidation()) {
+          disableSSLValidation();
+        }
+      }
+      conn.setRequestMethod(config.getMethod().toUpperCase());
+      conn.setConnectTimeout(config.getConnectTimeout());
+      conn.setReadTimeout(config.getReadTimeout());
+      conn.setInstanceFollowRedirects(config.getFollowRedirects());
+      conn.addRequestProperty("charset", config.getCharset());
+      for (Map.Entry<String, String> propertyEntry : headers.entrySet()) {
+        conn.addRequestProperty(propertyEntry.getKey(), propertyEntry.getValue());
+      }
+      //Default contentType value would be added in the request properties if user has not added in the headers.
+      if (config.getMethod().equals("POST") || config.getMethod().equals("PUT")) {
+        if (!headers.containsKey("Content-Type")) {
+          conn.addRequestProperty("Content-Type", contentType);
+        }
+      }
+      if (messages.length() > 0) {
+        conn.setDoOutput(true);
+        try (OutputStream outputStream = conn.getOutputStream()) {
+          outputStream.write(messages.toString().trim().getBytes(config.getCharset()));
+        }
+      }
+      //responseCode = conn.getResponseCode();
+      httpStatusCode = conn.getResponseCode();
+      LOG.debug("HTTP Status Code: {}", httpStatusCode);
+      messages.setLength(0);
+    } catch (MalformedURLException | ProtocolException e) {
+      throw new IllegalStateException("Error opening url connection. Reason: " + e.getMessage(), e);
+    } catch (IOException e) {
+      LOG.warn("Error making {} request to url {} with headers {}.", config.getMethod(), config.getUrl(), headers);
+    } finally {
+      if (conn != null) {
+        conn.disconnect();
+      }
+    }
+    RetryableErrorHandling errorHandlingStrategy = httpErrorHandler.getErrorHandlingStrategy(httpStatusCode);
+    return !errorHandlingStrategy.shouldRetry();
```

Comment:
Setting `messages.setLength(0)` before knowing `errorHandlingStrategy.shouldRetry()` will send a empty messege body when the retry call is made

Reply:
Shifted it elsewhere. 4e4ee6211be03c9e368a811866c8d96ec49518d3

Reply:
Setting is 0 for ever write, will clamp message size 1.
This is not compatible with batch logic hence when batchSize is more than 1 this will not write to output any records other than last few.